test/fixture.h

```diff
@@ -295,31 +295,31 @@ class TestInstance {
     std::unique_ptr<TestFixture> impl;
 };
 
-// TODO: most asserts do not actually assert i.e. do not return
 #define TEST_CASE( NAME ) do { if (prepareTest(#NAME)) { setVerbose(false); try { NAME(); teardownTest(); } catch (...) { assertNoThrowFail(__FILE__, __LINE__); } } } while (false)
-#define ASSERT( CONDITION ) if (!assert_(__FILE__, __LINE__, (CONDITION))) return
-#define ASSERT_LOC( CONDITION, FILE_, LINE_ ) (void)assert_(FILE_, LINE_, (CONDITION))
-#define CHECK_EQUALS( EXPECTED, ACTUAL ) (void)assertEquals(__FILE__, __LINE__, (EXPECTED), (ACTUAL))
+
+// TODO: the asserts do not actually assert i.e. do stop executing the test
```

Comment:
As this was handled inconsistently before and we we always aim to have passing tests I am not sure we actually need to implement this. Having a misnomer is always bad though.

If we want to actually implement this we should move the failure handling into a single function and probably need to throw some special exception.